subsys/fs/nvs/nvs_priv.h

```diff
@@ -18,55 +18,27 @@ extern "C" {
 #define NVS_ID_GT(a, b) (((a) > (b)) ? ((((a)-(b)) > 0x7FFF) ? (0):(1)) : \
 	((((b)-(a)) > 0x7FFF) ? (1):(0)))
 
+
+#define NVS_MAX_LEN 0x7ffd
+
 /*
  * Special id values
  */
-#define NVS_ID_EMPTY      0xFFFF
-#define NVS_ID_SECTOR_END 0xFFFE
+#define NVS_ID_EMPTY		0xFFFF
+#define NVS_ID_SECTOR_START	0xFFFE
+#define NVS_ID_SECTOR_END	0xFFFD
+
 /*
  * Status return values
  */
 #define NVS_STATUS_NOSPACE 1
 
-#define NVS_MOVE_BLOCK_SIZE 8
-
-struct nvs_entry {
-	off_t data_addr; /* address in flash to write data */
-	u16_t len;       /* entry length in bytes */
-	u16_t id;        /* entry id, is 0xFFFFF when empty, set to 0xFFFE
-			  * for the last entry in a sector
-			  */
-};
+#define NVS_BLOCK_SIZE 8
```

Comment:
This is basically the maximum length that will be passed to the driver API in a single call.
I can definitely see a use for this, and I would be great to let the user see it menuconfig.